Adafruit_MQTT.cpp

```diff
@@ -327,6 +327,47 @@ uint16_t Adafruit_MQTT::readFullPacket(uint8_t *buffer, uint16_t maxsize,
   return ((pbuff - buffer) + rlen);
 }
 
+#ifdef ARDUINO_ARCH_ESP32
+const char *Adafruit_MQTT::connectErrorString(int8_t code) {
+  const char *statusMsg;
+  switch (code) {
+  case 1:
+    statusMsg =
+        "The Server does not support the level of the MQTT protocol requested";
+    break;
+  case 2:
+    statusMsg =
+        "The Client identifier is correct UTF-8 but not allowed by the Server";
+    break;
+  case 3:
+    statusMsg = "The MQTT service is unavailable";
+    break;
+  case 4:
+    statusMsg = "The data in the user name or password is malformed";
+    break;
+  case 5:
+    statusMsg = "Not authorized to connect";
+    break;
+  case 6:
+    statusMsg = "Exceeded reconnect rate limit. Please try again later.";
+    break;
+  case 7:
+    statusMsg = "You have been banned from connecting. Please contact the MQTT "
+                "server administrator for more details.";
+    break;
+  case -1:
+    statusMsg = "Connection failed";
+    break;
+  case -2:
+    statusMsg = "Failed to subscribe";
+    break;
+  default:
+    statusMsg = "Unknown error";
+    break;
+  }
+  return statusMsg;
+}
+#else
```

Comment:
Would it work to just continue returning a `F()` for all cases? And then only add preproc around the function line to set the specific return type to match what `F()` ends up being. That would let the function code itself remain unchanged.

Reply:
@caternuson  I do not think so as `F()` is different return types. Could you give me a quick example of what you mean, having trouble visualizing this solution?

Reply:
Something like this maybe?
```cpp
#ifdef ARDUINO_ARCH_ESP32
const char *Adafruit_MQTT::connectErrorString(int8_t code) {
#else
const __FlashStringHelper *Adafruit_MQTT::connectErrorString(int8_t code) {
#endif
  switch (code) {
  case 1:
    return F(
        "The Server does not support the level of the MQTT protocol requested");
```

Basically just changing the return type as needed for whatever `F()` ends up being on a given platform.

Reply:
Implemented in https://github.com/adafruit/Adafruit_MQTT_Library/pull/222/commits/a94f4d512e966718f466d4b97ecb129a67e38347, requesting a re-review before I make the same change to the Arduino Adafruit IO Library.